### PR TITLE
Add TextInputFileRaw and TextOutputFileRaw

### PR DESCRIPTION
**Pull request by ChrisJefferson**
Add the ability to access files without compression, for packagemanager (and other users who might need this).

Needs some more tests before merging, checking if the naming / general design is reasonable / fixes the problem

### CODE REVIEW

**Comment by mtorpey** [comment]
Thanks for doing this, Chris!  This should make problems in curlInterface/PackageManager fixable.

**Review by mtorpey — APPROVED** [review]
This looks great to me!  Not sure why it's failing Travis checks, but as far as I'm concerned it could be merged.

File: lib/streams.gd [hunk 1]
```diff
@@ -623,12 +623,15 @@ DeclareOperation( "InputTextString", [ IsString ] );
...
 ##  <ManSection>
 ##  <Oper Name="InputTextFile" Arg='filename'/>
+##  <Oper Name="InputTextFileRaw" Arg='filename'/>
 ##
 ##  <Description>
 ##  <C>InputTextFile( <A>filename</A> )</C> returns an input stream in the category
 ##  <Ref Filt="IsInputTextStream"/> that delivers the characters from the file
 ##  <A>filename</A>. If <A>filename</A> ends in <C>.gz</C> and the file is
 ##  a valid gzipped file, then the file will be transparently uncompressed.
+##  <C>InputTextFileRaw</C> acts the same as <C>InputTextFile</>, without automatic
```
Comment (fingolfin) [inline lib/streams.gd:633]:
```suggestion
##  <C>InputTextFileRaw</C> acts the same as <C>InputTextFile</C>, without automatic
```

File: lib/streams.gd [hunk 2]
```diff
@@ -623,12 +623,15 @@ DeclareOperation( "InputTextString", [ IsString ] );
...
+##  uncompression.
```
Comment (fingolfin) [inline lib/streams.gd:634]:
```suggestion
##  decompression.
```

File: src/sysfiles.h [hunk 3]
```diff
@@ -114,7 +114,7 @@ void SyBufSetEOF(Int fid);
 **  Right now GAP does not read nonascii files, but if this changes sometimes
 **  'SyFopen' must adjust the mode argument to open the file in binary mode.
 */
-Int SyFopen(const Char * name, const Char * mode);
+Int SyFopen(const Char * name, const Char * mode, BOOL transcompress);
```
Comment (fingolfin) [inline src/sysfiles.h:117]: What does `transcompress` stand for? I would never guess what it does based on this name. Can you please also document it in the documentation comment above this function prototype?

**Review by fingolfin — CHANGES_REQUESTED** [review]

**Comment by fingolfin** [comment]
@ChrisJefferson ping? you might want to at least merge the fixes I suggested?

**Comment by ChrisJefferson** [comment]
Fixes applied, will be continued in #4128 , as I put this in gap-system/gap by accident.

**Pull request closed without merging**